ddtrace/contrib/celery/app.py

```diff
@@ -0,0 +1,44 @@
+from celery import signals
+
+from ddtrace import Pin
+from ddtrace.pin import _DD_PIN_NAME
+from ddtrace.ext import AppTypes
+
+from .util import APP, WORKER_SERVICE
+from .signals import (
+    trace_prerun,
+    trace_postrun,
+    trace_before_publish,
+    trace_after_publish,
+    trace_failure,
+)
+
+
+def patch_app(app, pin=None):
+    """Attach the Pin class to the application and connect
+    our handlers to Celery signals.
+    """
+    pin = pin or Pin(service=WORKER_SERVICE, app=APP, app_type=AppTypes.worker)
+    pin.onto(app)
+
+    signals.task_prerun.connect(trace_prerun)
```

Comment:
The same signal is not called multiple times so our handlers are called globally if at least one app is instrumented. In general, we have only one Celery app active for each process, even though we support multiple apps running at the same time.